src/backend/mod.rs

```
@@ -4870,14 +4880,25 @@ impl<'ctx> StreamOps for AudioUnitStream<'ctx> {
         self.stopped.store(false, Ordering::SeqCst);
         self.draining.store(false, Ordering::SeqCst);
 
-        // Execute start in serial queue to avoid racing with destroy or reinit.
-        let result = self
-            .queue
+        self.queue
             .clone()
-            .run_sync(|| self.core_stream_data.start_audiounits())
-            .unwrap();
-
-        result?;
+            .run_sync(|| -> Result<()> {
+                // Need reinitialization: device was changed when paused.
+                if self.delayed_reinit {
+                    self.reinit().inspect_err(|_| {
+                        cubeb_log!(
+                            "({:p}) delayed reinit during start failed.",
+                            self.core_stream_data.stm_ptr
+                        );
+                    })?;
+                    self.delayed_reinit = false;
+                }
+                // Execute start in serial queue to avoid racing with destroy or reinit.
+                self.core_stream_data.start_audiounits().inspect_err(|_| {
+                    cubeb_log!("({:p}) start failed.", self.core_stream_data.stm_ptr);
+                })
+            })
+            .ok_or(Error::error());
```

I think it's fine to `.unwrap()` here.

---

We should jus return an error no?

---

The only time `run_sync` [returns `None`](https://searchfox.org/mozilla-central/rev/0c55d51c0d2a9b672e42ad40ea54f90267f92a8e/third_party/rust/coreaudio-sys-utils/src/dispatch.rs#157,164-167,174) is if the queue is canceled (`run_final`'s task [has finished](https://searchfox.org/mozilla-central/rev/0c55d51c0d2a9b672e42ad40ea54f90267f92a8e/third_party/rust/coreaudio-sys-utils/src/dispatch.rs#198-200)). That is not supposed to happen here so better that we fail hard.